### DISCUSSION TITLE
Gaps in time series

### DISCUSSION THREAD

**Discussion by laurent-leballeur** [georgebv/pyextremes#93] [General]
Is pyextremes able to take account of gaps in the time series? to consider the actual duration of observations and the resulting return periods.
Thanks.
Regards

**Comment by georgebv** [comment]
I don't understand the question. How would one take gaps and duration into account in the context of EVA? I'm also not sure what gap and duration mean for irregular discrete time series.

**Comment by laurent-leballeur** [comment]
For example, if one performs a POT analysis on a discontinuous dataset (some periods when the measurement is unavailable), pyextremes seems to calculate the rate of extreme events as:
`n_periods = (ts.index.max() - ts.index.min()) / return_period_sizes`
The computation of n_periods does not consider times when data is not available

**Reply by georgebv** [reply]
I see. Here you can go in two directions: (a) assume there were no extremes in the gaps and (b) generate synthetic data.

For `BM` you can set `errors` to `coerce` to fill gaps with average extreme values (option b) or to `ignore` (option a).
For `POT` the default is option a. For option b you will need to generate your own synthetic data.
